docs/miscellaneous.rst

```diff
@@ -88,6 +88,234 @@ by checking if the lowest bit is set: short (not set) and long (set).
 .. note::
   Handling invalidly encoded slots is currently not supported but may be added in the future.
 
+JSON Output
+===========
+
+The storage layout of a contract can be requested via the :ref:`standard JSON interface <compiler-api>`.  The output is a JSON object containing two keys,
+``storage`` and ``types``.  The ``storage`` object is an array where each
+element has the following form:
+
+
+.. code::
+
+
+    {
+        "ast_id":"2",
+        "contract":"A",
+        "label":"x",
+        "offset":"0",
+        "slot":"0",
+        "type":"t_uint256"
+    }
+
+where the example above is the storage layout of ``contract A { uint x; }``
+and
+
+- ``ast_id`` is the id of the AST node of the state variable
+- ``contract`` is the name of the contract
+- ``label`` is the name of the state variable
+- ``offset`` is the offset within the storage slot according to the encoding
+- ``slot`` is the storage slot where the state variable resides
+- ``type`` is an identifier used as key to the variable's type information
+
+The given ``type``, in this case ``t_uint256`` represents an element in
+``types``, which has the form:
+
+
+.. code::
+
+    {
+        "encoding":"inplace",
+        "label":"uint256",
+        "numberOfBytes":"32",
+        "numberOfSlots":"1"
+    }
+
+where
+
+- ``encoding`` tells how the data is stored
```

Comment:
I'd say this is not very helpful and has to be expanded - it should list all possible values and what they mean exactly...

Reply:
This should be in the section right above this one that explains how the storage layout/encoding works.

Reply:
I can list the options linking them back to the section before.

Reply:
Yeah, an exhaustive list of possible values linking to the respective session should be enough, if it's sufficiently explained there.

Reply:
Updated